proposals/infra/INF-0002-root-signature-in-clang.md

```diff
@@ -304,15 +440,113 @@ elimation has completed.
 
 ### Validations in Sema
 
-TODO
+#### All the values should be legal.
+
+Some of these values are covered by syntactical checks in Sema.
+
+List all here for reference during DXIL generation checks.
+
+##### ShaderVisibility
+- SHADER_VISIBILITY_ALL
+- SHADER_VISIBILITY_ALL
+- SHADER_VISIBILITY_VERTEX
+- SHADER_VISIBILITY_HULL
+- SHADER_VISIBILITY_DOMAIN
+- SHADER_VISIBILITY_GEOMETRY
+- SHADER_VISIBILITY_PIXEL
+- SHADER_VISIBILITY_AMPLIFICATION
+- SHADER_VISIBILITY_MESH
+
+##### ParameterType
+- CBV
+- SRV
+- UAV
+- DescriptorTable
+- Constants32Bit
+
+##### RootDescriptorFlags
+- None
+- DataVolatile
+- DataStaticWihleSetAtExecute
+- DataStatic
+
+##### DescriptorRangeFlags
+
+###### Sampler
+- None
+- DescriptorsVolatile
+- DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS
+
+###### Resource
+- None
+- DESCRIPTORS_VOLATILE,
+- DATA_VOLATILE,
+- DATA_STATIC,
+- DATA_STATIC_WHILE_SET_AT_EXECUTE,
+- DESCRIPTORS_VOLATILE | DATA_VOLATILE,
+- DESCRIPTORS_VOLATILE | DATA_STATIC_WHILE_SET_AT_EXECUTE,
+- DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS,
+- DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS | DATA_VOLATILE,
+- DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS | DATA_STATIC,
+- DESCRIPTORS_STATIC_KEEPING_BUFFER_BOUNDS_CHECKS | DATA_STATIC_WHILE_SET_AT_EXECUTE,
+
+##### StaticSampler
+- Max/MinLOD cannot be NaN.
+- Comparison filter must have ComparisonFunc not equal to 0.
```

Comment:
The grammar has a list of valid ComparisonFuncs, so how would a user specify a syntactically correct StaticSampler is a ComparsionFunc equal to 0?

Reply:
For the backend, the ComparisonFunc is just an i32 which could be 0.

Reply:
This section is describing the validations performed in Sema.  I can't see any way in Sema for this to happen.  There's a separate section for describing the validations performed during codegen.

Reply:
Moved to DXIL generation part.